mantidproject/mantid
# CodeEditor auto-completion switch

This PR implements a request from an ILL user to have a switch that disables auto-completion suggestions in the `CodeEditor` widget of the workbench.

**To test:**

1. Code review
2. Testing:
- Open workbench, by default the suggestions should be present in the `CodeEditor` widget, so try typing python code (e.g. np.) and observe suggestions, try also with an algorithm from `simpleapi`
- Change the setting 'Provide auto-completion suggestions in the script editor' which can be found in `Settings/General/General` to `False`
- Try again to get suggestions by typing in the `CodeEditor` widget
- Open a new tab, it should behave in the same way as the previous tab, so there should be no suggestions.
- Restart workbench. Try to get suggestions from the `CodeEditor`, it should behave the same way as before closing. Check settings widget if the appropriate switch is still unchecked.

Fixes #32862 . 

Release notes will be added shortly.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
